### DISCUSSION TITLE
Clarify Alpine V3 integration

### DISCUSSION THREAD

**Discussion by mcrumm** [livewire/livewire#3279] [1. Help] [answered]
Hello everyone!  I am here on a diplomatic mission for the Phoenix LiveView docs :)

Recently a PR was opened to update the LiveView docs for users wishing to integrate Alpine V3. However [a question was raised](https://github.com/phoenixframework/phoenix_live_view/pull/1518#issuecomment-874332203) asking whether or not we should direct users to include a nodeType check as seen here:
https://github.com/livewire/livewire/blob/1ca6757c78dbead4db7f52a72dabb8b27efcb3f6/js/component/SupportAlpine.js#L287-L296

At a glance it appears to me that the nodeType check is redundant to the property check, but given that I am probably missing something, I thought I would ask here. Thanks!

**Comment by joshhanley** [comment, marked as answer]
@mcrumm welcome!

I've just had a look, and yeah I agree, not exactly sure the reason for the nodeType check here.

If you look at the Alpine V2 integration, it didn't have it, so I'd also be curious as to why it's used here.
https://github.com/livewire/livewire/blob/1ca6757c78dbead4db7f52a72dabb8b27efcb3f6/js/component/SupportAlpine.js#L248-L254

I will run some tests against LW without it in and see if anything fails, which might highlight its use.

Only person who would know definitely would be @calebporzio.

Hope this helps!

**Reply by joshhanley** [reply]
@mcrumm didn't seem to cause any failures in the tests when I removed it.

I think it is a morphdom thing anyway. Does LiveView use morphdom or something else?

**Reply by mcrumm** [reply]
@joshhanley Thanks for trying that out!

LiveView does use morphdom– that's what made especially curious about the extra nodeType check here in Livewire, since Alpine V3 has some of its own morph-related things now I want to make sure we're giving users a proper integration path. :)

**Reply by calebporzio** [reply]
Thanks for weighing in and verifying the tests @joshhanley - I don't remember off hand that that line is necessary - could very well be redundant - likely something left over from a previous attempt at something different.

Let me know what else we can do to make integration with LiveView easier @mcrumm!

**Reply by mcrumm** [reply]
Thanks @calebporzio, I appreciate the response!
